src/components/molecules/Visualizer/Plugin/hooks.ts

```diff
@@ -51,12 +53,14 @@ export default function ({
       ? `${pluginBaseUrl}/${`${pluginId}/${extensionId}`.replace(/\.\./g, "")}.js`
       : undefined;
 
-  const autoResize = useMemo((): "both" | "width-only" | "height-only" | undefined => {
-    if (!!widget?.extended?.horizontally && !!widget?.extended?.vertically) return "both";
-    if (widget?.extended?.horizontally) return "width-only";
-    if (widget?.extended?.vertically) return "height-only";
+  const autoResizeDirection = useMemo((): "both" | "width-only" | "height-only" | undefined => {
+    if (autoResize) return autoResize;
+    const { horizontally, vertically } = widget?.extended ?? {};
+    if (horizontally && vertically) return "both";
+    if (vertically) return "width-only";
+    if (horizontally) return "height-only";
     return undefined;
-  }, [widget?.extended?.horizontally, widget?.extended?.vertically]);
+  }, [widget?.extended, autoResize]);
```

Comment:
Since the computation time is O(1) (= since there is no loop processing, the processing time is always constant, no matter what time it is.), the process is not as heavy as using useMemo. For such a simple process, the overhead of useMemo will be larger.

```ts
const actualAutoResize = autoResize ?? (
	widget?.extended?. horizontally && widget.extended.vertically ? "both" :
		widget?.extended?.vertically ? "width-only" :
			widget?.extended?. horizontally);
```